### PR TITLE
feat: frontend webpack proxy support zstd encoding

### PR DESCRIPTION
**Pull request by rebareba**
### ADDITIONAL INFORMATION

- [ ] Has associated issue: #28366  #29678 
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by rusackas** [comment]
Thank you so much for this! Looks like it's working fine. It _might_ need a rebase to pick up a CI fix from another PR, and it looks like it has some linting issues, which can be resolved with our pre-commit hooks. 

Superset uses Git pre-commit hooks courtesy of [pre-commit](https://pre-commit.com/). To install run the following:
```
pip3 install -r requirements/development.txt
pre-commit install
A series of checks will now run when you make a git commit.
```
Alternatively it is possible to run pre-commit by running pre-commit manually:
```
pre-commit run --all-files
```

File: superset-frontend/webpack.proxy-config.js [hunk 1]
```diff
@@ -17,6 +17,7 @@
  * under the License.
  */
 const zlib = require('zlib');
+const {ZSTDDecompress} = require('simple-zstd');
```
Comment (rusackas) [inline superset-frontend/webpack.proxy-config.js:20]:
```suggestion
const { ZSTDDecompress } = require('simple-zstd');
```

**Comment by rusackas** [comment]
Pushed a commit to (hopefully) fix the linting issue

**Comment by rusackas** [comment]
Not sure why Docker is failing here (cc @mistercrunch )

Good news is, `npm run dev-server` works as expected again, hooray!

When I go to port 8088, however, it doesn't look like the webpack build happened, and the frontend isn't present.

**Comment by mistercrunch** [comment]
@dosubot what is zstd and why do we need it?

**Comment by mistercrunch** [comment]
> Not sure why Docker is failing here (cc @mistercrunch )

I think we're hitting memory limits in GHA, it seems we've been right against the edge on it and I'm afraid PRs are going to fail randomly more and more until we address it. I discussed it here https://github.com/apache/superset/pull/29771#issuecomment-2261357135, and I'm a bit at a loss for a solution.

**Comment by rebareba** [comment]
Docker failed This looks like we needs to install zstd in System. The npm package `simple-zstd ` use system command zstd to compression and decompression .
`apt-get -y install zstd`

**Comment by rusackas** [comment]
Superseded by https://github.com/apache/superset/pull/30034
Thanks for figuring this out @rebareba !

**Pull request closed without merging**